### PR TITLE
Crps noise update

### PR DESCRIPTION
**Pull request by nadiyashore**
What’s Changed:

- Added CRPS calculation to Generator loss function in gan_gp.py
- CRPS definition in losses.py
- Updated documentation / comments to reflect inclusion of CRPS in model-performance reporting.
- Removed the noise layer from the network architecture (models.py line 279)

### CODE REVIEW

File: ClimatExML/conf/config.yaml
Comment (SBeairsto) [inline ClimatExML/conf/config.yaml]: These changes to `config.yaml` look like formatting for this particular use case. Generally we avoid updating these unless there is a more general reason to do so. And in the interest of clearly documenting updates to the code, we want to keep any code updates in this pull request focused on the CRPS implementation. Any non-CRPS related changes will want to be in a separate pull request. (I know this seems slow and clunky, but it's very handy when you are looking back through code versions trying to figure out where a specific change was made).

#### Recommendation:
Omit these changes to `config.yaml` in the pull request. You can keep a copy of this version _locally_ to run your implementation with.

File: ClimatExML/models.py [hunk 1]
```diff
@@ -276,7 +276,7 @@ def __init__(
         self.conv3 = nn.Sequential(
             # nn.Conv2d(filters * 2, filters + 1, kernel_size=3, stride=1, padding=1),
             nn.Conv2d(filters * 2, filters + 1, kernel_size=3, stride=1, padding=1),
-            ResidualInResidualDenseBlock(filters + 1, noise, resolution=fine_dims),
+            # ResidualInResidualDenseBlock(filters + 1, noise, resolution=fine_dims), noise layer removed due to "bubles"/noise artifacts
```
Comment (SBeairsto) [inline ClimatExML/models.py]: Soooo, this should probably be a separate pull request, as it does not directly pertain to the CRPS implementation and _is_ an important bug fix that we want properly documented.  

### Recommendation:
__Ideally__: Move this to another pull request (so remove it from this branch, create new branch based off main and add it there, create pull request) and in the pull request description outline the bug you were seeing including an image(s). You could additionally add an issue in the GitHub issues tab describing the bug. 

__Alternatively__ (keep it simple this time): update the name of this pull request to include the bug fix, and have a separate section in the pull request description briefly describing the bug (including image). Again, ChatGPT can be helpful to come up with a quick blurb.

File: ClimatExML/wgan_gp.py [hunk 2]
```diff
@@ -94,11 +95,25 @@ def training_step(self, batch, batch_idx):
...
         go_downhill(self, loss_c, c_opt)
 
+        #copied over from wgan_gp_stoch.py 
+        # the n realisation part and crps part are not in wgan_gp.py
         if (batch_idx + 1) % self.n_critic == 0:
             self.toggle_optimizer(g_opt)
             sr = self.G(lr, hr_cov)
-            loss_g = -torch.mean(self.C(sr).detach()) + self.alpha * mean_squared_error(sr, hr)
-            go_downhill(self, loss_g, g_opt)
+            n_realisation = 4 # related to batch size needs to be egual in config
```
Comment (SBeairsto) [inline ClimatExML/wgan_gp.py]: I will take a look at this and see if I can have it tied to configs. Generally, we want to avoid hard coded values in our code (parameters defined outside of `config.yaml`).

**Review by SBeairsto — COMMENTED** [review]
Great job Nadyia! One or two things to review vis-a-vis best practices, but this will be a super valuable addition to the main branch!
